app-modules/knowledge-base/resources/views/filament/pages/badges.blade.php

```diff
@@ -33,7 +33,32 @@
 --}}
 
 <div class="flex gap-2">
+    @if ($getRecord()->status)
+        <x-filament::badge>
+            {{ $getRecord()?->status?->name ?? '' }}
+        </x-filament::badge>
+    @endif
     <x-filament::badge>
-        {{ $getRecord()?->status?->name ?? '' }}
+        {{ !empty($getRecord()['public']) ? 'Public' : 'Internal' }}
     </x-filament::badge>
+    @if (!empty($getRecord()['public']) && !empty($getRecord()['category_id']))
```

Comment:
Is it possible to keep the way we access attributes on a record consistent? So as properties instead of as array keys? `$getRecord()->public`

Reply:
No, we cannot do like that
Code Checks / fix-code-style(PR check)  giving error for that because i think public is reserve keyword so.

Reply:
I think now it will not give because we are using not empty i think. I write wrong syntax i think that's why it giving. below is error i got.

Error:  app-modules/knowledge-base/resources/views/filament/pages/badges.blade.php: SyntaxError: Parse Error : syntax error, unexpected 'public' (T_PUBLIC) on line 1
Error:  > 1 | <?php echo $getRecord()?->public ?? false ? 'Public' : 'Internal' ?>

I will add now like you suggest because we are now using !empty so.

Reply:
To be honest I think `public` should not have existed as an attribute in the first place, regardless of its keyword status, it should be `is_public` to match the name of other booleans. Thoughts on the best way to correct this @orrison?

Reply:
@danharrin specifically to address whether or not we want to rename the column. I think I agree, but it would beyond the scope of this ticket to do so.
I will add a ticket to our backlog to discuss this and eventually make the change.

@ketan-canyon are you sure the lining issue is because of a reserved keyword issue? I may be wrong but I have a suspicion that this is not the case.
It is likely that linting is throwing that error because static analysis does not know what `$getRecord()` actually returns in this case.
You can fix this by setting `$getRecord()` to a variable and adding a `@var` docblock for that variable to type hint it to the correct Model.

Reply:
@Orrison 
Not lint, its Code Checks / fix-code-style checking giving me syntax error. but now we are using !empty so, i will check again this and update here